### PR TITLE
Enable prow jobs for `iam-controller`

### PR DESCRIPTION
**Pull request by A-Hilaly**
Issue if available: N/A

Description of changes:
-  Enable Prow tests and management for the new iam-controller repository.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prow/jobs/jobs_config.yaml [hunk 1]
```diff
@@ -21,6 +21,7 @@ aws_services:
 - sagemaker
 - sfn
 - sns
+- iam
```
Comment (jaypipes) [inline prow/jobs/jobs_config.yaml:24]: I think these are in alphabetical order?

Comment (A-Hilaly) [reply prow/jobs/jobs_config.yaml]: ++ changed

**Review by jaypipes — APPROVED** [review]
:+1:

**Pull request merged**